lightning/src/util/ser_macros.rs

```diff
@@ -165,6 +165,10 @@ macro_rules! decode_tlv {
 	}};
 }
 
+/// Decode a TLV stream that contains custom TLVs that are unknown to LDK but may be known to the
+/// user.
+macro_rules! decode_tlv_stream_with_custom {}
```

Comment:
Just to check - because onion message TLVs are split into "control types" and "message types", this is split, too? ie it takes a list of TLVs to decode like we normally decode TLVs, and then also reads anything higher than a certain constant as "custom TLVs".

Reply:
Also, is it at all practical to have some kind of matching on the onion type and use a different decoder per-type? Would be really nice to not have to allocate a vec of TLV decode entries for onion message decodes we implement ourselves.